### PR TITLE
Fix 132

### PR DESCRIPTION
**Pull request by kascheri12**

### CODE REVIEW

**Comment by LefterisJP** [comment]
@kascheri12 is this ready for review?

**Comment by kascheri12** [comment]
Yes. I think it's still WIP, but please review and provide direction.

File: ui/dashboard.js [hunk 1]
```diff
@@ -78,10 +78,32 @@ function create_box (id, icon, number, currency_icon) {
     settings.page_index = $('#page-wrapper').html();
 }
 
+function update_eth_node_conn_status_ui(is_local_eth) {
```
Comment (LefterisJP) [inline ui/dashboard.js]: please use full names not abbreviation. So here: `update_eth_node_connection_status_ui`

File: ui/dashboard.js [hunk 2]
```diff
@@ -78,10 +78,32 @@ function create_box (id, icon, number, currency_icon) {
...
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-link fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-success"><p>Connected to a local ethereum node</p></ul>';
+    }
+    $('.eth-node-status').html(str);
+}
 
 
 var alert_id = 0;
 function add_alert_dropdown(alert_text, alert_time) {
+    if (
+        alert_text.indexOf("ethereum") > 0
```
Comment (LefterisJP) [inline ui/dashboard.js]: Why have 2 different checks here for the same thing? Also having them all inside the `if()` makes it harder to read. A nice way to improve readability here would be:

```javascript
let connection_fail = alert_text.indexOf('Could not connect to a local ethereum node') >= 0;
let no_sync = alert_text.indexOf('Could not transact with/call contract function') >= 0;
if (connection_faill || no_sync) {
    update_eth_node_connection_status_ui(false);
} else {
    update_eth_node_connection_status_ui(true);
}
```

File: ui/dashboard.js [hunk 3]
```diff
@@ -78,10 +78,32 @@ function create_box (id, icon, number, currency_icon) {
...
+    } else {
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-link fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-success"><p>Connected to a local ethereum node</p></ul>';
+    }
+    $('.eth-node-status').html(str);
+}
 
 
 var alert_id = 0;
 function add_alert_dropdown(alert_text, alert_time) {
+    if (
```
Comment (LefterisJP) [inline ui/dashboard.js]: also for the future: I guess there is no other way right now to see if we are connected to an ethereum node, but reading the alert text is hacky. Should properly expose it from the backend.

**Comment by LefterisJP** [comment]
So @kascheri12 left some comments in the code. Also checked it locally and wanted to say it looks nice and close to what I imagined. Have some feedback on the positioning. I am attaching a screenshot to point to where I think it should be located.

rotkehlchen_location_of_link_notification

In the future I want this, the last time balances were save and other notification icons that may be added in some kind of dock there on the top left. This dock should start where the line separating the left menu from the dashboard is located.

File: ui/dashboard.js (hunk 3 above)
Comment (kascheri12) [reply ui/dashboard.js]: Yes, I understand and agree. How would I go about doing that?

**Comment by kascheri12** [comment]
@LefterisJP ready for review.

File: ui/dashboard.js [hunk 4]
```diff
@@ -78,10 +78,26 @@ function create_box (id, icon, number, currency_icon) {
     settings.page_index = $('#page-wrapper').html();
 }
 
+function update_eth_node_connection_status_ui(is_local_eth) {
+    let str;
+    if (!is_local_eth) {
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-unlink fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-warning"><p>Not connected to a local ethereum node</p></ul>';
```
Comment (LefterisJP) [inline ui/dashboard.js]: Here please remove the `alert-warning`. Gives a different colour to the popup.

File: ui/dashboard.js [hunk 5]
```diff
@@ -78,10 +78,26 @@ function create_box (id, icon, number, currency_icon) {
...
+    } else {
+        str='<a class="eth-node-status-link" href="#"><i class="fa fa-link fa-fw"></i></a><ul class="dropdown-menu dropdown-eth-status alert-success"><p>Connected to a local ethereum node</p></ul>';
```
Comment (LefterisJP) [inline ui/dashboard.js]: Here please remove the alert-success class. Gives a different color to the popup.

**Comment by kascheri12** [comment]
Hi @LefterisJP, thank you for your consistent review and the chance to collaborate on this project!

**Pull request merged**